src/payment/strategies/cybersource/cybersource-payment-strategy.ts

```diff
@@ -0,0 +1,266 @@
+import { includes, some } from 'lodash';
+import { Subject } from 'rxjs/index';
+import { take } from 'rxjs/operators';
+
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import {
+    MissingDataError, MissingDataErrorType, NotInitializedError, NotInitializedErrorType,
+    RequestError
+} from '../../../common/error/errors';
+import InvalidArgumentError from '../../../common/error/errors/invalid-argument-error';
+import { OrderActionCreator } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import OrderRequestBody from '../../../order/order-request-body';
+import PaymentMethodCancelledError from '../../errors/payment-method-cancelled-error';
+import { CreditCardInstrument } from '../../payment';
+import PaymentActionCreator from '../../payment-action-creator';
+import PaymentMethod from '../../payment-method';
+import PaymentMethodActionCreator from '../../payment-method-action-creator';
+import { PaymentInitializeOptions, PaymentRequestOptions } from '../../payment-request-options';
+import PaymentStrategy from '../payment-strategy';
+
+import {
+    default as SignatureValidationErrors,
+    CardinalEventAction, CardinalEventResponse, CardinalEventType, CardinalInitializationType, CardinalPaymentBrand,
+    CardinalPaymentStep,
+    CardinalTriggerEvents,
+    CardinalValidatedAction, CardinalValidatedData, CyberSourceCardinal, SetupCompletedData,
+} from './cybersource';
+import CyberSourceScriptLoader from './cybersource-script-loader';
+
+export default class CyberSourcePaymentStrategy implements PaymentStrategy {
+    private _Cardinal?: CyberSourceCardinal;
+    private _paymentMethod?: PaymentMethod;
+    private _cardinalEvent$: Subject<CardinalEventResponse>;
+    private _isSetupCompleted = false;
+    private _cardinalSessionId?: string;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _CyberSourceScriptLoader: CyberSourceScriptLoader
+    ) {
+        this._cardinalEvent$ = new Subject();
+    }
+
+    initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId } = options;
+
+        if (this._isSetupCompleted) {
+            Promise.resolve(this._store.getState());
+        }
+
+        return this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod('cybersourcedirect')).then( state => {
```

Comment:
We are working on `bcapp` in order to refactor the `getProvider` end point in order to be able to call it with `cybersource (methodId)`  instead of `cybersourcedirect`, this bug is due to `bcapp` is not mapping providers correctly from storefront